git/revlist.go

```diff
@@ -8,8 +8,11 @@ import (
 // List of command line options that may be passed to RevList
 type RevListOptions struct {
 	Quiet, Objects bool
+	MaxNumCommits  *int
```

Comment:
I think we should stick to the same terminology as the command line ("MaxCount"), because the terminology MaxNumCommits isn't accurate when Objects is true.

Also, I think it it should be either `*uint` or `int` (with a convention of a negative number means "all".. I think I prefer the latter.)

Reply:
Sure, I'll change the name. The git docs here talk about number commits specifically so I thought that it was always about commits: https://git-scm.com/docs/git-rev-list

I was going to use -1 as the convention but then I was thinking that the pointer makes better use of the zero value as the default (show me everything). If I switch it to *uint it will also prevent any really nonsensical negative numbers. I'll switch it to *uint unless you have reservations.

Reply:
The only reservation I have against `*uint` is that then you've always gotta check for nil before dereferencing, while `int` is only one check, either make sense to me.